libcore/src/geometry/helper/CorrectGeometry.cpp

```diff
@@ -0,0 +1,354 @@
+        return std::nullopt;
+    }
+
+    // Intersection points with NAN cannot be split
+    if(std::isnan(intersectionPoint._x) || std::isnan(intersectionPoint._y)) {
+        return std::nullopt;
+    }
+
+    Logging::Debug(fmt::format(
+        check_fmt("BigWall {} will be split due to intersection with Line {} in {}"),
+        wall.toString(),
+        line.toString(),
+        intersectionPoint.toString()));
+
+    return {intersectionPoint};
+}
+
+/**
+ * @brief Split a wall in several small walls
+ *
+ * search all walls+crossings+transitions that intersect <bigwall>
+ * not in an endpoint
+ *
+ * @param subroom: subroom containing <bigwall>
+ * @param bigWall: wall to split
+ * @return std::vector: a vector of all small walls. Can be empty.
+ */
+std::optional<std::vector<Wall>> SplitWall(const SubRoom & subroom, const Wall & bigWall)
+{
+    Logging::Debug(fmt::format(
+        check_fmt("SplitWall for BigWall {} in SubRoom {}-{}."),
+        bigWall.toString(),
+        subroom.GetRoomID(),
+        subroom.GetSubRoomID()));
+
+    // Collecting all split Points
+    std::vector<Point> splitPoints;
+    for(const auto & crossing : subroom.GetAllCrossings()) {
+        if(auto p = GetSplitPoint(bigWall, *crossing)) {
+            splitPoints.emplace_back(*p);
+        }
+    }
+    for(const auto & transition : subroom.GetAllTransitions()) {
+        if(auto p = GetSplitPoint(bigWall, *transition)) {
+            splitPoints.emplace_back(*p);
+        }
+    }
+    for(const auto & wall : subroom.GetAllWalls()) {
+        if(auto p = GetSplitPoint(bigWall, wall)) {
+            splitPoints.emplace_back(*p);
+        }
+    }
+
+    // No split points found, bigWall cannot be split.
+    if(splitPoints.empty()) {
+        return std::nullopt;
+    }
+
+    // Sort the splitpoints by distance to bigWall Point1
+    std::sort(
+        splitPoints.begin(), splitPoints.end(), [&bigWall](const Point & p1, const Point & p2) {
+            return Distance(bigWall.GetPoint1(), p1) < Distance(bigWall.GetPoint1(), p2);
+        });
+
+    std::vector<Wall> wallPieces;
+    // We are not interested in all wall segments since we only want to remove outstanding segments
+    if(splitPoints.size() == 1) {
+        wallPieces.emplace_back(bigWall.GetPoint1(), splitPoints.front());
+        wallPieces.emplace_back(splitPoints.back(), bigWall.GetPoint2());
+    } else {
+        wallPieces.emplace_back(bigWall.GetPoint1(), splitPoints.front());
+        wallPieces.emplace_back(splitPoints.front(), splitPoints.back());
+        wallPieces.emplace_back(splitPoints.back(), bigWall.GetPoint2());
+    }
+
+    return wallPieces;
+}
+
+bool isPointAndSubroomIncident(const SubRoom & subroom, const Point & point)
+{
+    // Incident with wall at point 1 or 2
+    for(const auto & checkWall : subroom.GetAllWalls()) {
+        if(checkWall.GetPoint1() == point || checkWall.GetPoint2() == point) {
```

Comment:
This is a floating point comparison right? Should this be rather an approx. comparison?

Reply:
The equal operator of `Point` considers an macro defined epsilon.